src/collection.jl

```diff
@@ -130,15 +225,167 @@ end
 
 
 """
-    images(df::DataFrame)
+    ccds(df::DataFrame)
 
-Generator for `ImageHDU`s of entries in data frame.
+Generator for `CCDData`s of entries in data frame.
+
+Iterates over `collection` using each `path` and `hdu` to load data using [`FITSIO.FITS`](http://juliaastro.github.io/FITSIO.jl/latest/api.html#FITSIO.FITS) into a [`CCDData`](@ref).
+
+# Examples
+```julia
+collection = fitscollection("~/data/tekdata")
+for hdu in ccds(collection)
+    @assert hdu isa CCDData
+end
+```
 """
-function images end
+function ccds end
 
-# generator for ImageHDU specified by data frame (i.e. path of file, hdu etc.)
-@resumable function images(df::DataFrame)
+# generator for CCDData specified by data frame (i.e. path of file, hdu etc.)
+@resumable function ccds(df::DataFrame)
     for row in eachrow(df)
-        @yield FITS(row.path)[row.hdu]
+        @yield CCDData(row.path; hdu = row.hdu)
     end
 end
+
+
+"""
+    ccds(f, collection::DataFrame; save = false, path = nothing, save_prefix = nothing, save_suffix = nothing, save_delim = "_", ext = r"fits(\\.tar\\.gz)?"i, kwargs...)
+
+Iterates over the `CCDData`s of the collection applying function `f` at each step.
+
+It returns an array of output values of function `f` applied on `CCDData`s.
+In addition to applying function `f`, the outputs can be saved. If `save = true`, it enables programmatical saving of returned value of the function `f` using [`CCDReduction.write_data`](@ref). File is saved at `path` specified by the user.
+Suffix and prefix can be added to filename of newly created files by modifying `save_suffix` and `save_prefix`, `save_delim` is used as delimiter.
+`ext` is the extension of files in collection, by default it is set to `r"fits(\\.tar\\.gz)?"i`.
+
+# Example
+```julia
+collection = fitscollection("~/data/tekdata")
+processed_images = map(ccds(collection)) do img
+    trim(img, (:, 1040:1059))
+end
+```
+The above generates `processed_images` which consists of trimmed versions of images present in `collection`.
+
+For saving the `processed_images` simultaneously with the operations performed
+```julia
+processed_images = map(ccds(collection; save = true, path = "~/data/tekdata", save_prefix = "trimmed")) do img
+    trim(img, (:, 1040:1059))
+end
+```
+The trimmed images are saved as `trimmed_(original_name)` (FITS files) at `path = "~/data/tekdata"` as specified by the user.
+
+Mapping version of `ccds` function is interfaced on iterative version of `images`, any valid parameter can be passed into iterative version as `kwargs`.
+"""
+function ccds(f, collection::DataFrame; save = false, path = nothing, save_prefix = nothing, save_suffix = nothing, save_delim = "_", ext = r"fits(\.tar\.gz)?"i, kwargs...)
+    image_iterator = ccds(collection; kwargs...)
+    locations = collection.path
+
+    processed_images = map(zip(locations, image_iterator)) do (location, output)
+        processed_image = f(output)
+        if save
```

Comment:
I'm torn here-

on one hand, it's _very_ clear and _very_ simple to only save via a keyword argument. However, consider `ccds(pipelin, coll; save_suffix="_cal")`. This won't actually save the files, and I now have to manually add `save=true` even though I may have assumed that it _would_ save!

I would lean towards some automatic `save` behavior that can be overridden instead of requiring it. Something like
```julia
save = any(!isnothing, (save_prefix, path, save_suffix))
```

Reply:
So, should we remove the `save` keyword from the list of input parameters?

PS: Keeping it would make sure that we will be saving (definitely sure!)
Whereas it is possible that a user may set `save_prefix` without actually wanting to save it.

PS: I don't think that makes a lot of sense.
Let's remove save from keywords, what do you say?

Reply:
If you make save a keyword with a default value (like I just showed you) means that it will behave "automatically" but the user can still override it. Without the keyword argument the user can't override it.